src/common/components/MultiSelect.jsx

```diff
@@ -22,6 +25,15 @@ export default function MultipleSelectCheckmarks({
   label,
 }) {
   const classes = useStyles();
+  const [searchText, setSearchText] = useState("");
+
+  const filteredOptions = useMemo(
+    () =>
+      options.filter((option) =>
+        compareTextValue(option.label, searchText, typeof option.label)
+      ),
+    [searchText, options]
+  );
```

Comment:
what is the purpose of useMemo? what improvements it is bringing?

Reply:
I am re-calculating the filteredOptions only when the `searchText` or the `options` array changes. In this way, `options.filter` wouldn't be invoked when any other state variable causes a re-render. This is my understanding. Let me know if there is a better way to achieve this mechanism.

Reply:
yup. actually `useMemo` is not meant to use this way. for the current requirements we can have `useEffect` for that and moreover also `useEffect` doesnt make sense anywhere when we put object as dependency. its same as creating a normal function and invoke it while mapping over it (since we are mapping over `filteredOptions`). for now you can go with useEffect or a function whicever seems easier. 
if you got confused make a `console.log` inside the useMemo and see it runs on each renders.